src/main/java/com/gregtechceu/gtceu/api/machine/multiblock/WorkableMultiblockMachine.java

```diff
@@ -269,4 +274,24 @@ public void setWorkingEnabled(boolean isWorkingAllowed) {
     public GTRecipeType getRecipeType() {
         return recipeTypes[activeRecipeType];
     }
+
+    public GTRecipe.ActionResult input(boolean simulate, Map<RecipeCapability<?>, List<Content>> contents){
+        GTRecipe tmp=new GTRecipe(new GTRecipeType(new ResourceLocation("gt_machine_io"),"gt"),
+            new ResourceLocation("___recipe_test_ids__"),contents,null,null,
+            null,null,List.of(),null,0,false);
+        if(simulate)return tmp.matchRecipe(IO.IN,this,contents,false);
+        else return tmp.handleRecipe(IO.IN,this,contents)?GTRecipe.ActionResult.SUCCESS: GTRecipe.ActionResult.FAIL_NO_REASON;
+    }
+    public GTRecipe.ActionResult output(boolean simulate, Map<RecipeCapability<?>, List<Content>> contents){
+        GTRecipe tmp=new GTRecipe(new GTRecipeType(new ResourceLocation("gt_machine_io"),"gt"),
+            new ResourceLocation("___recipe_test_ids__"),null,contents,null,
+            null,null,List.of(),null,0,false);
+        if(simulate)return tmp.matchRecipe(IO.OUT,this,contents,false);
+        else return tmp.handleRecipe(IO.OUT,this,contents)?GTRecipe.ActionResult.SUCCESS: GTRecipe.ActionResult.FAIL_NO_REASON;
+    }
```

Comment:
hell naw part 3

Reply:
also, again, formatting.

Reply:
？

Reply:
same thing as above. this is janky, just plain wrong, why would you do this?

Reply:
So what should I do?

Reply:
...not have that.

Reply:
This is used for dynamically inputting and outputting to the machine. Do you have any better suggestions?

Reply:
making a recipe, or an addon.
or waiting for Ichor (KJS's new javascript engine that's in the works) to be published, because that can create classes.

Reply:
I think input and output should be handled within the machine, rather than in the recipe.
You might consider refactoring this part of the input and output.